### PR TITLE
docs: document the return type restrictions for operator overloading

### PR DESCRIPTION
**Pull request by thesombady**
Documentation for operator overloading with return type limitation.

### CODE REVIEW

**Review by Delta456 — APPROVED** [review]
Good catch

**Comment by Delta456** [comment]
I will be doing a follow-up PR to make it a checker error when this is merged.

File: doc/docs.md [hunk 1]
```diff
@@ -6191,6 +6191,7 @@ To improve safety and maintainability, operator overloading is limited.
 
 - When overriding `<` and `==`, the return type must be strictly `bool`.
 - Both arguments must have the same type (just like with all operators in V).
+- Overloaded operators has to return the same type as the argument (with the expection `<` and `==` which returns boolean expressions).
```
Comment (Delta456) [inline doc/docs.md:6194]: As this line goes beyond the 100 words. I think

```suggestion
- Overloaded operators have to return the same type as the argument (with the exception `<` and `==).
```

works better

Comment (thesombady) [reply doc/docs.md:6194]: Wow, trying to write to fast hehe :/

**Pull request merged**